app/src/main/java/protect/card_locker/MainActivity.java

```diff
@@ -741,4 +747,26 @@ public void onRowClicked(int inputPosition)
             startActivityForResult(i, Utils.MAIN_REQUEST);
         }
     }
+
+    @Override
+    protected void onPostResume() {
+        super.onPostResume();
+
+        settings = new Settings(this);
+        if(settings.getAppLockStatus()){  // App Lock On
+            new UnlockDialogBuilder(this)
+                    .onUnlocked(() -> {
+                        appIsLocked = false;
+                        updateLoyaltyCardList();
+                        Toast.makeText(this, "Unlocked", Toast.LENGTH_SHORT).show();
+                    })
+                    .onCanceled(() -> { })
+                    .showIfRequiredOrSuccess(TimeUnit.MINUTES.toMillis(0));
+
+        } else{
+            appIsLocked = false;
+            updateLoyaltyCardList();
+        }
+
+    }
```

Comment:
It only protects MainActivity, so widgets are not protected and neither is the Settings activity, allowing you to disable this by directly going to the settings activity (I recommend moving the unlock to CatimaAppCompatActivity, from which all activities inherit).

Reply:
Then it would probably make sense to create an `OnAppUnlocked` function in CatimaAppCompatActivity, which is called after unlocking or calling it instantly if there is no lock.

That way, each activity that needs to change something on unlocking can override the `OnAppUnlocked` function.

The default implementation in CatimaAppCompatActivity should simply do nothing.

Reply:
I also suggest creating a `isActivityLockable` function that returns true by default in CatimaAppCompatActivity. That way we can opt out some stuff (like: not require unlocking if the Edit Activity is opened from the "Create new card" flow)